app/common/components/table.test.js

```diff
@@ -0,0 +1,179 @@
+    });
+  }));
+
+  it('should render <div> for data with no href property', createTestHarness(setup, (harness) => {
+    const context = {
+      params: {
+        ...mockContext.params,
+        data: [
+          [{ data: 'Greater Manchester CCG' }, { data: 'X01' }],
+          [{ data: 'Hampshire CCG' }, { data: 'X02' }],
+        ],
+      },
+    };
+
+    harness.request(context, ($) => {
+      context.params.data.forEach((row, rowIndex) => {
+        row.forEach((dataPoint, i) => {
+          const divTag = $(`[data-test-id="table-row-${rowIndex}"] div:nth-child(${i + 1})`);
+          expect(divTag.text().trim()).toEqual(dataPoint.data);
+          expect(divTag.hasClass(mockContext.params.columnClass)).toEqual(true);
+          expect($(`[data-test-id="table-row-${rowIndex}"] a`).length).toEqual(0);
+        });
+      });
+    });
+  }));
+
+  it('should render <a> for data with href property', createTestHarness(setup, (harness) => {
+    const context = {
+      params: {
+        ...mockContext.params,
+        data: [
+          [
+            { data: 'Greater Manchester CCG', href: 'organisations/org1' },
+            { data: 'X01', href: 'ods/X01' },
+          ],
+          [
+            { data: 'Hampshire CCG', href: 'organisations/org2' },
+            { data: 'X02', href: 'ods/X02' },
+          ],
+        ],
+      },
+    };
+
+    harness.request(context, ($) => {
+      context.params.data.forEach((row, rowIndex) => {
+        row.forEach((dataPoint, i) => {
+          const aTag = $(`[data-test-id="table-row-${rowIndex}"] a:nth-child(${i + 1})`);
+          expect(aTag.text().trim()).toEqual(dataPoint.data);
+          expect(aTag.hasClass(mockContext.params.columnClass)).toEqual(true);
+          expect(aTag.attr('href')).toEqual(dataPoint.href);
+          expect($(`[data-test-id="table-row-${rowIndex}"] div`).length).toEqual(0);
+        });
+      });
+    });
+  }));
+
+  it('should render tag component for columns with tag property', createTestHarness(setup, (harness) => {
+    const context = {
+      params: {
+        ...mockContext.params,
+        data: [
+          [
+            {
+              tag: {
+                dataTestId: 'a-tag-id-1',
+                text: 'tag text',
+                classes: 'a-class',
+              },
+            },
+            {
+              tag: {
+                dataTestId: 'a-tag-id-2',
+                text: 'tag text 2',
+                classes: 'a-class-2',
+              },
+            },
+          ],
+          [
+            {
+              tag: {
+                dataTestId: 'a-tag-id-3',
+                text: 'tag text 3',
+                classes: 'a-class-3',
+              },
+            },
+            {
+              tag: {
+                dataTestId: 'a-tag-id-4',
+                text: 'tag text 4',
+                classes: 'a-class-4',
+              },
+            },
+          ],
```

Comment:
Do we need all 4? can we get away with just 1. You've already proven out that it shows all the rows and columns. Just need to prove it renders the tag component when specified